client/angular/components/thread_page/revision_history_modal/revision_history_modal.coffee

```diff
@@ -1,45 +1,42 @@
-Records = require 'shared/services/records.coffee'
-
-{ applyLoadingFunction } = require 'shared/helpers/apply.coffee'
+Records           = require 'shared/services/records.coffee'
 
 angular.module('loomioApp').factory 'RevisionHistoryModal', ->
   templateUrl: 'generated/components/thread_page/revision_history_modal/revision_history_modal.html'
-  controller: ['$scope', 'model', ($scope, model) ->
+  controller:['$scope', 'model', ($scope,  model) ->
     $scope.model = model
-    $scope.loading = true
-
-    $scope.load = ->
-      switch $scope.model.constructor.singular
-        when 'discussion' then Records.versions.fetchByDiscussion($scope.model.key)
-        when 'comment'    then Records.versions.fetchByComment($scope.model.id)
-
-    $scope.header =
-      switch $scope.model.constructor.singular
-        when 'discussion' then 'revision_history_modal.thread_header'
-        when 'comment'    then 'revision_history_modal.comment_header'
-
-    $scope.discussionRevision = ->
-      $scope.model.constructor.singular == 'discussion'
-
-    $scope.commentRevision = ->
-      $scope.model.constructor.singular == 'comment'
-
-    $scope.threadTitle = (version) ->
-      $scope.model.attributeForVersion('title', version)
-
-    $scope.revisionBody = (version) ->
-      switch $scope.model.constructor.singular
-        when 'discussion' then $scope.model.attributeForVersion('description', version)
-        when 'comment'    then $scope.model.attributeForVersion('body', version)
-
-    $scope.threadDetails = (version) ->
-      if version.isOriginal()
-        'revision_history_modal.started_by'
-      else
-        'revision_history_modal.edited_by'
-
-    applyLoadingFunction($scope, 'load')
-    $scope.load()
-
-    return
-  ]
+    $scope.type = model.constructor.singular
+    $scope.currentIndex = latestIndex = model.versionsCount - 1
+
+    $scope.getVersion = () ->
+      Records.versions.fetchVersion($scope.model, $scope.currentIndex).then (data) ->
+        $scope.version = Records.versions.find(data.versions[0].id)
+        if $scope.version.changes.group_id
+          Records.groups.findOrFetchById($scope.version.changes.group_id[1])
+
+    $scope.isOldest = ->
+      $scope.currentIndex == 0;
+
+    $scope.isNewest = ->
+      $scope.currentIndex == latestIndex
+
+    $scope.setNextRevision = ->
+      if !$scope.isNewest()
+        $scope.currentIndex += 1;
+        $scope.getVersion()
+
+    $scope.setPreviousRevision = ->
```

Comment:
I'd love to cut down on these methods some. It seems like we could have something like

```
$scope.displayVersion = (index) ->
  $scope.currentIndex = index
  $scope.getVersion()
```
then, in the haml:
```
.md-button{ng-if: "currentIndex > 0", ng-click: "displayVersion(0)"} # display the first version
.md-button{ng-if: "currentIndex > 0", ng-click: "displayVersion(currentIndex-1)"} # display the previous version
.md-button{ng-if: "currentIndex < maxIndex", ng-click: "displayVersion(currentIndex+1)"} # display the next version
.md-button{ng-if: "currentIndex < maxIndex", ng-click: "displayVersion(maxIndex)"} # display the last version
```